### PR TITLE
t3c: strategies.yaml hash_key only for consistent_hash

### PR DESCRIPTION
**Pull request by traeak**
The t3c strategies.yaml file when generating a strategy always populates the hash_key yaml field although it only applies to consistent_hash policy.

Currently with this configuration ATS 9.2 will bump into the unexpected hash_key field for non consistent_hash policy and abort loading the strategies file entirely.

Note the ATS docs here where hash_key only applies to consistent_hash policy.
https://docs.trafficserver.apache.org/en/latest/admin-guide/files/strategies.yaml.en.html?highlight=strategies#strategies-definitions

<hr/>

## Which Traffic Control components are affected by this PR?

- Traffic Control Cache Config (`t3c`, formerly ORT)

## What is the best way to verify this PR?

Create DS services, one with consistent_hash and another with round robin.
Run t3c to generate strategies.yaml
Ensure that the DS with consistent_hash policy also has a hash_key key/value
Ensure that the DS with round robin (rr_strict or rr_ip) does not have a hash_key entry.

As a bonus, drop the strategies.yaml file into an ats config directory, fire up ATS9.2 and verify that the strategies.yaml file loads without generating an error.

## If this is a bugfix, which Traffic Control versions contained the bug?

7.1.0

## PR submission checklist
- [x] This PR has tests
- [ ] ~~This PR has documentation~~ bug fix
- [x] This PR has a CHANGELOG.md entry 
- [x] This PR **DOES NOT FIX A SERIOUS SECURITY VULNERABILITY** (see [the Apache Software Foundation's security guidelines](https://apache.org/security) for details)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -55,8 +55,11 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 - [#7137](https://github.com/apache/trafficcontrol/pull/7137) *Traffic Control Cache Config (t3c)* parent.config simulate topology for non topo delivery services.
 - [#7153](https://github.com/apache/trafficcontrol/pull/7153) *Traffic Control Cache Config (t3c)* Adds an extra T3C check for validity of an ssl cert (crash fix).
 - [#3965](https://github.com/apache/trafficcontrol/pull/3965) *Traffic Router* Traffic Router now always includes a `Content-Length` header in the response.
+<<<<<<< HEAD
```
Comment (jpappa200) [inline CHANGELOG.md:58]: This looks like it's left over from a conflict.

**Review by jpappa200 — APPROVED** [review]
This looks good to me, just need to remove the typo in CHANGELOG.md

File: CHANGELOG.md (hunk 1 above)
Comment (traeak) [reply CHANGELOG.md]: removed!

File: CHANGELOG.md [hunk 2]
```diff
@@ -58,6 +58,8 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 - [#3965](https://github.com/apache/trafficcontrol/pull/3965) *Traffic Router* Traffic Router now always includes a `Content-Length` header in the response.
 - [#7182](https://github.com/apache/trafficcontrol/pull/7182) Sort peers used in strategy.yaml to prevent false positive for reload.
 - [#6533](https://github.com/apache/trafficcontrol/issues/6533) *TR should not rename/recreate log files on rollover
+- [#7182](https://github.com/apache/trafficcontrol/pull/7182) *Traffic Control Cache Config (t3c)* Sort peers used in strategy.yaml to prevent false positive for reload.
```
Comment (ocket8888) [inline CHANGELOG.md:61]: This line duplicates one that's 2 above it

**Review by ocket8888 — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 2 above)
Comment (ocket8888) [reply CHANGELOG.md:61]: But it does it better, by calling out the affected component, so it's probably best to keep your line and delete the old one.

**Review by ocket8888 — APPROVED** [review]

**Pull request merged**